### DISCUSSION TITLE
Two more functions for full-blooded JSON component

### DISCUSSION THREAD

**Discussion by stefanov-sm** [sqlpage/SQLPage#315] [Ideas]
### Motivation  
Build REST services that support `GET, POST, PUT, DELETE, ...` request methods and arbitrary length raw JSON input  

### Suggestion - add two more functions

- `sqlpage.request_method()` function that returns the request method as text. This would allow for code like

```sql
select case sqlpage.request_method()
  when 'GET' then (...)
  when 'POST' then (...)
  when 'PUT' then (...)
 ...
end;
```

- `sqlpage.request_body()` function that returns the request body payload as raw text. This would allow for code like (PostgreSQL)

```sql
with t(j) as 
(
  select jsonb_array_elements(sqlpage.request_body()::jsonb)
)
...
```
### Expected result
A fully functional REST services development toolset based on the JSON component.

**Comment by stefanov-sm** [comment]
One more thing is missing, how to set the HTTP response status code. Probably an extra optional top-level parameter of the JSON component - `response_status` with default value 200 if omitted?

**Comment by lovasoa** [comment]
Both additions would be very welcome! Do you want to try your hand at a small pull request? The function system has been reworked in 0.20.5 so adding new functions should be easier than ever now.

**Reply by stefanov-sm** [reply]
Thanks a lot! I would love to but I am afraid that I am ignorant in Rust and its ecosystem ...  I will have a look though.
